File: apps/api/AssetManagement.Api/DTOs/ReportDtos.cs
namespace AssetManagement.Api.DTOs;

// Asset Summary Report
public record AssetSummaryReportDto(
    int TotalAssets,
    decimal TotalValue,
    List<StatusBreakdownItemDto> ByStatus,
    List<AssetsByGroupItemDto> ByType,
    List<AssetsByGroupItemDto> ByLocation
);

// Combined Expiry Report
public record ExpiryItemDto(
    Guid Id,
    string Name,
    string Category,       // "Warranty", "Certificate", "Licence"
    string TypeName,       // asset type / cert type / app type
    DateTime ExpiryDate,
    int DaysUntilExpiry,
    string Status
);

public record ExpiriesReportDto(
    List<ExpiryItemDto> Items,
    int TotalCount
);

// Licence Summary Report
public record LicenceSummaryReportDto(
    int TotalApplications,
    int Active,
    int Expired,
    int PendingRenewal,
    int Suspended,
    decimal TotalSpend,
    List<LicenceExpiryItemDto> ExpiringSoon
);

// Assignments Report
public record PersonAssignmentDto(
    Guid PersonId,
    string FullName,
    string? Email,
    int AssignedAssetCount,
    List<AssignedAssetBriefDto> Assets
);

public record AssignedAssetBriefDto(
    Guid Id,
    string Name,
    string AssetTag
);

public record AssignmentsReportDto(
    int TotalAssigned,
    int TotalPeople,
    List<PersonAssignmentDto> People
);

// Asset Lifecycle Report
public record AssetLifecycleReportDto(
    List<AssetsByAgeBucketDto> ByAge,
    List<WarrantyExpiryItemDto> PastWarranty,
    List<OldestAssetDto> OldestAssets
);

public record OldestAssetDto(
    Guid Id,
    string Name,
    string AssetTag,
    string AssetTypeName,
    DateTime PurchaseDate,
    int AgeDays
);

File: apps/api-kt/src/main/kotlin/com/assetmanagement/api/dto/ReportDtos.kt
package com.assetmanagement.api.dto

import java.math.BigDecimal
import java.time.Instant
import java.util.*

data class AssetSummaryReportDto(
    val totalAssets: Int,
    val totalValue: BigDecimal,
    val byStatus: List<StatusBreakdownItemDto>,
    val byType: List<AssetsByGroupItemDto>,
    val byLocation: List<AssetsByGroupItemDto>
)

data class ExpiryItemDto(
    val id: UUID,
    val name: String,
    val category: String,
    val typeName: String,
    val expiryDate: Instant,
    val daysUntilExpiry: Int,
    val status: String
)

data class ExpiriesReportDto(
    val items: List<ExpiryItemDto>,
    val totalCount: Int
)

data class LicenceSummaryReportDto(
    val totalApplications: Int,
    val active: Int,
    val expired: Int,
    val pendingRenewal: Int,
    val suspended: Int,
    val totalSpend: BigDecimal,
    val expiringSoon: List<LicenceExpiryItemDto>
)

data class PersonAssignmentDto(
    val personId: UUID,
    val fullName: String,
    val email: String?,
    val assignedAssetCount: Int,
    val assets: List<AssignedAssetBriefDto>
)

data class AssignedAssetBriefDto(
    val id: UUID,
    val name: String
)

data class AssignmentsReportDto(
    val totalAssigned: Int,
    val totalPeople: Int,
    val people: List<PersonAssignmentDto>
)

data class AssetLifecycleReportDto(
    val byAge: List<AssetsByAgeBucketDto>,
    val pastWarranty: List<WarrantyExpiryItemDto>,
    val oldestAssets: List<OldestAssetDto>
)

data class OldestAssetDto(
    val id: UUID,
    val name: String,
    val assetTypeName: String,
    val purchaseDate: Instant,
    val ageDays: Int
)

data class DepreciationReportDto(
    val totalOriginalCost: BigDecimal,
    val totalAccumulatedDepreciation: BigDecimal,
    val totalBookValue: BigDecimal,
    val groups: List<DepreciationGroupDto>
)

data class DepreciationGroupDto(
    val assetTypeName: String,
    val subtotalOriginalCost: BigDecimal,
    val subtotalAccumulatedDepreciation: BigDecimal,
    val subtotalBookValue: BigDecimal,
    val assets: List<DepreciationAssetDto>
)

data class DepreciationAssetDto(
    val id: UUID,
    val name: String,
    val assetTypeName: String,
    val purchaseDate: Instant?,
    val originalCost: BigDecimal,
    val depreciationMethod: String,
    val usefulLifeYears: Int?,
    val accumulatedDepreciation: BigDecimal,
    val currentBookValue: BigDecimal,
    val remainingUsefulLifeMonths: Int?
)
